### DISCUSSION TITLE
Dynamic Segment Ignored

### DISCUSSION THREAD

**Discussion by allan-vera** [TanStack/router#929] [Q&A] [answered]
So I practically am replicating the example from https://tanstack.com/router/v1/docs/guide/route-paths#dynamic-segments, but my application just goes to the `/users` route even with my `users/$userId` route defined.

Here's my code:

```
const usersRoute = new Route({
  getParentRoute: () => rootRoute,
  path: "users",
  component: UsersPage,
  beforeLoad: (data) => protectRoute(data),
});
export const userDetailRoute = new Route({
  getParentRoute: () => usersRoute,
  path: `$userId`,
  component: UserDetailPage,
  beforeLoad: (data) => protectRoute(data),
});

export const routeTree = rootRoute.addChildren([
  usersRoute.addChildren([userDetailRoute]),
]);
```

`protectRoute` just double checks to make sure a user is logged in.

Devtools:
![Screenshot 2024-01-04 at 5 48 06 PM](https://github.com/TanStack/router/assets/153534041/97120116-87d3-44a8-b68b-7ac6c699e554)

**Comment by allan-vera** [comment]
Btw this alternative way works as suggested by https://tanstack.com/router/v1/docs/guide/route-paths#inner-path-slashes. Just not sure if this ought to be the way I do it given the previous link's docs.

```
const usersRoute = new Route({
  getParentRoute: () => rootRoute,
  path: "users",
  component: UsersPage,
  beforeLoad: (data) => protectRoute(data),
});
export const userDetailRoute = new Route({
  getParentRoute: () => rootRoute,
  path: "users/$userId",
  component: UserDetailPage,
  beforeLoad: (data) => protectRoute(data),
});

export const routeTree = rootRoute.addChildren([
  usersRoute,
  userDetailRoute,
]);
```

This alternative way

**Comment by schiller-manuel** [comment]
You probably should define a separate `usersIndexRoute`, similarly to `postsIndexRoute` in this example: https://tanstack.com/router/v1/docs/examples/react/basic

**Reply by allan-vera** [reply]
Yeah, I saw that and had that previously, but didn't make a difference so I removed it.

**Reply by schiller-manuel** [reply]
please provide a complete example which includes the index route for further analysis.

**Reply by allan-vera** [reply]
Here's my example:

```
const usersRoute = new Route({
  getParentRoute: () => rootRoute,
  path: "users",
  component: UsersPage,
  beforeLoad: (data) => protectRoute(data),
});
export const userDetailRoute = new Route({
  getParentRoute: () => usersRoute,
  path: "$userId",
  component: UserDetailPage,
  beforeLoad: (data) => protectRoute(data),
});
export const userIndexRoute = new Route({
  getParentRoute: () => usersRoute,
  path: "/",
  component: UserDetailPage,
  beforeLoad: (data) => protectRoute(data),
});

export const routeTree = rootRoute.addChildren([
  usersRoute.addChildren([userIndexRoute, userDetailRoute]),
]);


```
This is me at "users/2"
![Screenshot 2024-01-04 at 7 26 02 PM](https://github.com/TanStack/router/assets/153534041/8ca44620-8867-4384-8a92-bd0e5eecc49f)

**Reply by allan-vera** [reply]
And this is when I am at "users/"
![Screenshot 2024-01-04 at 7 26 54 PM](https://github.com/TanStack/router/assets/153534041/27fbc151-7239-498d-8622-6983bce5a77b)

**Reply by allan-vera** [reply]
![Screenshot 2024-01-04 at 7 30 07 PM](https://github.com/TanStack/router/assets/153534041/bbc23dd9-8b86-4bb5-816d-4be569dd2cb5)
Not sure how much it matters but I've noticed that in both scenarios (dynamic segments and inner path slashes) My images ( `<Image src={"imgs/example.svg"} width="auto" height={12} />` )  break.

**Reply by schiller-manuel** [reply, marked as answer]
Can you put the example up on e.g. codesandbox or a git repo? Trying to guess from some screenshots is not so great.
However, it looks like the routing works as expected?
`users/2` => `users/$userID`
`users/` => `users/`

What's the actual issue?
Is it that `UserDetailPage` is used in both routes? 
How about:

```tsx
const usersRoute = new Route({
  getParentRoute: () => rootRoute,
  path: "users",
});
export const userDetailRoute = new Route({
  getParentRoute: () => usersRoute,
  path: "$userId",
  component: UserDetailPage,
  beforeLoad: (data) => protectRoute(data),
});
export const userIndexRoute = new Route({
  getParentRoute: () => usersRoute,
  path: "/",
  component: UsersPage,
  beforeLoad: (data) => protectRoute(data),
});
```

About your images: You probably need absolute paths, e.g. `"/imgs/example.svg"` (leading slash!)

**Reply by allan-vera** [reply]
I could. But the issue is that UserRouter component (UsersPage) is appearing for both "$userId" and "/" routes. Essentially any component I place for those routes (or any child route for that matter) does not get rendered and instead is shown the UserRoute component.

**Reply by allan-vera** [reply]
@schiller-manuel I hadn't seen your proposed solution in my email inbox when I first reviewed it. But yeah, your implementation fixed it for me. Thank you!
